### PR TITLE
Fix credential interactions

### LINKED ISSUE #916: VFSForGit does not erase credentials on Authentication failures
VFSForGit has logic to erase and re-acquire crendentials if VFSForGit fails to authenticate against the service, but this appears to not be working.

Initial investigation indicates the problem is that VFSForGit made a change to use the full URL when acquiring credentials `usehttppath=true`, but this same setting is not used when erasing credentials.

### PR DESCRIPTION
**Pull request by jamill**
In testing, there were 2 changes necessary to reject credentials:

1) Make sure the `usehttppath` option is specified, so that the credential lookup and rejection use the same context

2) The call to reject credentials with username / password was not rejecting the credential (with GCM 1.18.5).

Additionally, since `usehttppath` only needs to be applied to `https://dev.azure.com` urls, we scope this setting. This should improve the experience when other remote URLs are used.

Fixes #916, #928  
Related: #949

### CODE REVIEW

File: GVFS/GVFS.Common/Git/GitProcess.cs [hunk 1]
```diff
@@ -179,7 +179,7 @@ public virtual void RejectCredentials(string repoUrl, string username = null, st
             string stdinConfig = sb.ToString();
 
             this.InvokeGitOutsideEnlistment(
-                "credential reject",
+                GenerateCredentialVerbCommand("reject"),
```
Comment (jeschu1) [inline GVFS/GVFS.Common/Git/GitProcess.cs]: How easy/hard would it  be to add unit tests in this space?

Comment (jeschu1) [reply GVFS/GVFS.Common/Git/GitProcess.cs]: @mjcheetham hahaha perfect :-)

**Comment by jamill** [comment]
@turbonaitis - this should address the issues you brought up in #949

File: GVFS/GVFS.Common/Git/GitProcess.cs [hunk 2]
```diff
@@ -760,6 +755,11 @@ public Process GetGitProcess(string command, string workingDirectory, string dot
             }
         }
 
+        private static string GenerateCredentialVerbCommand(string verb)
+        {
+            return $"-c {GitConfigSetting.CredentialUseHttpPath}=true credential {verb}";
```
Comment (kewillford) [inline GVFS/GVFS.Common/Git/GitProcess.cs]: Why are we not just setting this in the git config for the repo?  It shouldn't affect other git commands right? And should only be used for `dev.azure.com`?  Maybe we should check the repo url on clone and mount and set it in the config?  How will this affect clones done against GitHub?

Comment (jamill) [reply GVFS/GVFS.Common/Git/GitProcess.cs]: Thanks Kevin -

> Why are we not just setting this in the git config for the repo?

1) It is set on mount
2) Not all config commands that deal with credentials are run in the context of a repository.

> And should only be used for dev.azure.com

It is scoped to dev.azure.com, so will only affect dev.azure.com urls. Maybe tweaking the variable name would make this more clear? There could be a discussion about whether we should even set this if we know the URL is not a dev.azure.com URL, but:

1) It doesn't hurt to always set this (git will take it and only apply it to the expected urls)
2) It is arguably more consistent to always apply this in VFS4G
3) It is closer to where we want to end eventually

Comment (kewillford) [reply GVFS/GVFS.Common/Git/GitProcess.cs]: Okay thanks Jameson.  Can we just make add this to the const and not have this helper method like so
`public const string CredentialUseHttpPath = "-c credential.\"https://dev.azure.com\".useHttpPath=true";`
And then use a
`$"{CredentialUseHttpPath} credential fill"` etc in the invocations of git?

File: GVFS/GVFS.UnitTests/Git/GitAuthenticationTests.cs [hunk 3]
```diff
@@ -262,19 +262,19 @@ private MockGitProcess GetGitProcess()
             int approvals = 0;
             int rejections = 0;
             gitProcess.SetExpectedCommandResult(
-                "-c credential.useHttpPath=true credential fill",
+                "-c credential.\"https://dev.azure.com\".useHttpPath=true credential fill",
```
Comment (kewillford) [inline GVFS/GVFS.UnitTests/Git/GitAuthenticationTests.cs]: Add `const` for `-c credential.\"https://dev.azure.com\".useHttpPath=true` to use in all the tests?

File: GVFS/GVFS.Common/Git/GitProcess.cs (hunk 2 above)
Comment (jamill) [reply GVFS/GVFS.Common/Git/GitProcess.cs]: I like this suggestion, but we also use the the setting name directly in `TrySetRequiredGitConfigSettings`:

```
{ GitConfigSetting.CredentialUseHttpPath, "true" },
```

Comment (kewillford) [reply GVFS/GVFS.Common/Git/GitProcess.cs]: Then how about another `const` for `CommandLineDevAzureCredentialUseHttpPath` that is
`$"-c {CredentialUseHttpPath}=true";` or similar?

**Pull request merged**